src/App.tsx

```diff
@@ -225,27 +234,42 @@ export default class App extends React.Component<{}, { expandFolderBrowser: bool
     }
   }
 
-  onOkClick = async (ev: Event) => {
-    const selectedNode = (ev as CustomEvent<TreeNode>).detail;
-    const breadcrumbs = this.folderBrowser.current?.breadcrumbs;
-    const entryId = Number.parseInt(selectedNode.id, 10);
+  onSelectFolder = async () => {
+    if (!this.repoClient) {
+      throw new Error('Repo Client is undefined.');
+    }
+    if (!this.loginComponent) {
+      throw new Error('LfLoginComponent is not found.');
+    }
+    const selectedNode = this.repositoryBrowser.current?.currentFolder as LfRepoTreeNode;
+    let entryId = Number.parseInt(selectedNode.id, 10);
     const path = selectedNode.path;
-    const selectedFolder: LfFolder = {
+    if (selectedNode.entryType === EntryType.Shortcut) {
+      if (selectedNode.targetId)
+      entryId = selectedNode.targetId;
+    }
+    let selectedFolder = {
       entryId,
       path,
       displayName: this.getFolderNameText(entryId, path),
       displayPath: this.getFolderPathTooltip(path)
     };
-    if (breadcrumbs) {
-      selectedFolder.breadcrumbs = breadcrumbs;
-    }
     const nodeId = selectedNode.id;
-    const repoId = (await this.repoClient?.getCurrentRepoId());
-    const waUrl = this.loginComponent.current?.account_endpoints?.webClientUrl;
-    this.selectedNodeUrl = getEntryWebAccessUrl(nodeId, repoId!, waUrl!, selectedNode.isContainer);
-    this.selectedFolderDisplayName = selectedFolder?.displayName;
+    const repoId = (await this.repoClient.getCurrentRepoId());
+    const waUrl = this.loginComponent.current!.account_endpoints!.webClientUrl;
+    this.selectedNodeUrl = getEntryWebAccessUrl(nodeId, repoId, waUrl, selectedNode.isContainer);
     this.setState(() => { return { expandFolderBrowser: false, selectedFolder: selectedFolder } });
+    this.setState(() => { return { shouldShowOpen: this.setShouldShowOpen() }});
+    this.setState(() => { return { shouldShowSelect: this.setShouldShowSelect() }});
+  }
+
+  onEntrySelected = (event: any) => {
+    const treeNodesSelected: LfTreeNode[] = event.detail;
+    this.entrySelected = treeNodesSelected?.length > 0 ? treeNodesSelected[0] : undefined;
+    this.setState(() => { return { shouldShowOpen: this.setShouldShowOpen() }});
```

Comment:
there should just be one setState call
this.setState(() => { return { shouldShowOpen: this.setShouldShowOpen(). shouldShowSelect: this.setShouldShowSelect() }});